src/routing/Outlet.ts

```diff
@@ -1,100 +1,97 @@
-import { DNode, WidgetBaseInterface } from '../widget-core/interfaces';
+import { DNode } from '../widget-core/interfaces';
 import { WidgetBase } from '../widget-core/WidgetBase';
-import { w } from '../widget-core/d';
-import { inject } from '../widget-core/decorators/inject';
 import { alwaysRender } from '../widget-core/decorators/alwaysRender';
-import { OnEnter, Component, OutletOptions, OutletComponents, Outlet, Params, OutletContext } from './interfaces';
+import { MatchDetails, OnExit, OutletContext, OnEnter, Params } from './interfaces';
 import { Router } from './Router';
+import { diffProperty } from '../widget-core/decorators/diffProperty';
+import { Handle } from '../core/Destroyable';
 
-export function isComponent<W extends WidgetBaseInterface>(value: any): value is Component<W> {
-	return Boolean(value && (typeof value === 'string' || typeof value === 'function' || typeof value === 'symbol'));
+export interface OutletProperties {
+	renderer: (matchDetails: MatchDetails) => DNode;
```

Comment:
Can this not return an array of `DNodes` ?

Reply:
I think it should be able to.